src/misc_early.rs

```diff
@@ -113,4 +124,57 @@ impl EarlyLintPass for MiscEarly {
             }
         }
     }
+
+    fn check_expr(&mut self, cx: &EarlyContext, expr: &Expr) {
+        if let ExprKind::Call(ref paren, _) = expr.node {
+            if let ExprKind::Paren(ref closure) = paren.node {
+                if let ExprKind::Closure(_, ref decl, ref block) = closure.node {
+                    span_lint_and_then(cx,
+                                       REDUNDANT_CLOSURE_CALL,
+                                       expr.span,
+                                       "Try not to call a closure in the expression where it is declared.",
+                                       |db| {
+                                            if decl.inputs.len() == 0 {
+                                                let hint = format!("{}", snippet(cx, block.span, ".."));
+                                                db.span_suggestion(expr.span, "Try doing something like: ", hint);
+                                            }
+                                        });
+                }
+            }
+        }
+    }
+
+    fn check_block(&mut self, cx: &EarlyContext, block: &Block) {
+        for w in block.stmts.windows(2) {
+            if let StmtKind::Decl(ref first, _) = w[0].node {
+                if let DeclKind::Local(ref local) = first.node {
+                    if let Option::Some(ref t) = local.init {
+                        if let ExprKind::Closure(_,_,_) = t.node {
+                            if let PatKind::Ident(_,sp_ident,_) = local.pat.node {
+                                let mut arg_name = sp_ident.node.to_string();
```

Comment:
Just directly use sp_ident.node.to_string() below. That way you can use if_let_chain for this whole thing.